server/src/main/java/io/crate/metadata/cluster/DropTableClusterStateTaskExecutor.java

```diff
@@ -48,12 +54,25 @@ public DropTableClusterStateTaskExecutor(MetadataDeleteIndexService deleteIndexS
     @Override
     protected ClusterState execute(ClusterState currentState, DropTableRequest request) throws Exception {
         RelationName relationName = request.tableIdent();
-        Set<Index> concreteIndices = Set.of(IndexNameExpressionResolver.concreteIndices(
-            currentState.metadata(),
-            IndicesOptions.LENIENT_EXPAND_OPEN,
-            relationName.indexNameOrAlias()));
+        Metadata currentMetadata = currentState.metadata();
 
-        currentState = deleteIndexService.deleteIndices(currentState, concreteIndices);
+        @Nullable
+        RelationMetadata relation = currentMetadata.getRelation(relationName);
+        Collection<Index> indices;
+        if (relation instanceof BlobTable table) {
```

Comment:
In the other branch there is a `metadata.getIndices` which will likely replace the `IndexNameExpressionResolver.concreteIndices`. See:
 https://github.com/crate/crate/blob/506aacef6e58b71a33a94d8d479239f20119c568/server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java#L1248-L1271

I could already add this here as well - I think if we used this as a bridge and change all the `IndexNameExpressionResolver` calls it will make the migration to the new structures a lot easier.

One of the problems I faced with the other branch is that as soon as I add the rename/swap based on UUID all the resolve logic still based on index names breaks.

Reply:
Using a bridge upfront sounds good.

Reply:
Pushed a fixup.